packages/zoe/zoe.js

```diff
@@ -48,6 +48,49 @@ const makeZoe = (additionalEndowments = {}) => {
   const getAssaysFromPayoutRules = payoutRules =>
     payoutRules.map(payoutRule => payoutRule.units.label.assay);
 
+  const depositPayments = (offerRules, offerPayments) => {
+    const assays = getAssaysFromPayoutRules(offerRules.payoutRules);
+
+    // Promise flow = assay -> purse -> deposit payment -> escrow receipt
+    const paymentDepositedPs = assays.map((assay, i) => {
+      const assayRecordP = assayTable.getPromiseForAssayRecord(assay);
+      const payoutRule = offerRules.payoutRules[i];
+      const offerPayment = offerPayments[i];
+
+      return assayRecordP.then(({ purse, unitOps }) => {
+        if (
+          payoutRule.kind === 'offerExactly' ||
+          payoutRule.kind === 'offerAtMost'
+        ) {
+          // We cannot trust these units since they come directly
+          // from the remote assay and must coerce them.
+          return E(purse)
+            .depositExactly(payoutRule.units, offerPayment)
+            .then(units => unitOps.coerce(units));
+        }
+        insist(
+          offerPayments[i] === undefined,
+        )`payment was included, but the rule kind was ${payoutRule.kind}`;
+        return Promise.resolve(unitOps.empty());
+      });
+    });
+
+    return Promise.all(paymentDepositedPs).then(unitsArray => {
+      const extentsArray = unitsArray.map(units => units.extent);
+      const offerImmutableRecord = {
+        instanceHandle: undefined,
```

Comment:
Should `instanceHandle` still be undefined when it comes from the facet?

I noticed this only because you hadn't yet consolidated `escrowEmptyOffer`, so I was checking it for equivalence. It does initialize `instanceHandle`.

Also, just checking: When we get rid of receipts, then the `instanceHandle` will always be initialized before inserting the offerRecord into the table, right?

Reply:
Ah, good point. `instanceHandle` should be defined if the call is made from the smart contract. I changed `depositPayments` to take an optional `instanceHandle`

Reply:
And, yes! When we get rid of receipts, there is no escrow before choosing a contract instance to join.